File: jan15/jan15.js
function calculateFactorial(n) {
    if (n === 0 || n === 1) {
        return 1;
    } else {
        return n * calculateFactorial(n - 1);
    }
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question('Enter a number to calculate its factorial: ', (number) => {
        const n = parseInt(number);
        if (isNaN(n) || n < 0) {
            console.log("Please enter a valid non-negative integer.");
        } else {
            const result = calculateFactorial(n);
            console.log(`The factorial of ${n} is: ${result}`);
        }
        rl.close();
    });
}

main();

File: jan15/jan15.py
m = int(input("Enter an integer: "))
n = m
factorial = 1

if m == 0:
  print("0! = 1")
  quit()
  
else:
  while m > 0:
    factorial *= m
    m -= 1
    
    
    
print(f"{n}! = {factorial}")